Updated the README example with acquire connection

nkconnor opened the pull request:
Initially from reading the docs and examples I tried to use `&mut pool` instead of `&mut conn`. The compiler gave me an error that `Pool<MySql>` didn't implement `Executor`. I had to do a bit of digging and eventually just viewed the source of `Pool` to find `acquire()`, `try_acquire()` etc.

I think this change makes it a bit easier for someone to get started.

abonander commented:
In most cases we don't expect the user to have to explicitly grab a connection.

The other examples instead should be changed to use `&mut &pool` (because `&Pool` implements `Executor` and the methods take `&mut Executor`... we're not exactly happy with this situation but it's a breaking change so we're saving it for 0.3).

nkconnor commented:
doesn't this mean any wrapper struct methods would have to take `&mut self?`

e.g. 

```
async fn select_mysql_data(pool: &mut MySqlPool) -> {
  sqlx::query!(..)
    .fetch_one(pool)
    ...
} 

struct DataService{
  mysql:MySqlPool, // needs to be owned to share mutable ref
  cassandra:&Session
}

impl DataService {
   fn get_data(&self) -> () {
       select_mysql_data(self.pool).await // oops can't borrow as mutable, this is behind a shared &
   }
}
```

that's kinda rough to integrate into some existing APIs. is that the part you're changing in 0.3?

abonander commented:
Notice it's `&mut (&pool)`, so you don't need mutable access to the pool; you're passing a mutable reference to the immutable reference, which is perfectly legal (semantically this means the immutable reference could be set to point to a different pointer value with the same lifetime). This is how [`impl std::io::Read for &[u8]`](https://doc.rust-lang.org/std/io/trait.Read.html#impl-Read-10) works; the slice is updated to point to the unread bytes.

Our 0.3 change would be to make all methods take `Executor` by-value and then implement `Executor` for `&Pool` and `&mut MySqlConnection`/`&mut PgConnection`/`&mut PoolConnection` so you don't need that extra weird borrow with `Pool`. The current approach made sense to us initially (we don't need nor want ownership of an `Executor` for any methods) but we've found that the double-ref is really unintuitive for most people.

nkconnor commented:
I updated the README with `&mut &pool` and also removed the `tx = ` style from a couple examples. Note I had to make a few other changes to get examples to even compile, including changing from `&str` to `String` on the description paramater in todos, as well as dropping trailing `,` in macros, which I think is referenced in another PR or issue

nkconnor commented:
I was also curious about `Cargo.lock` in regards to the examples. Not sure if this should be updated?

abonander commented:
We prefer rebasing rather than merging `master` into branches; it keeps the history cleaner.

mehcode commented:
Thanks for the adjustments. Merging so things are easier to reference for 0.2.
